modules/apps/headless/headless-admin-workflow/headless-admin-workflow-test/src/testIntegration/java/com/liferay/headless/admin/workflow/resource/v1_0/test/WorkflowDefinitionLinkResourceTest.java

```diff
@@ -102,16 +110,85 @@ protected WorkflowDefinitionLink randomWorkflowDefinitionLink()
 				WorkflowDefinitionLink workflowDefinitionLink)
 		throws Exception {
 
+		return workflowDefinitionLinkResource.
+			postWorkflowDefinitionByExternalReferenceCodeWorkflowDefinitionLink(
+				_kaleoDefinition.getExternalReferenceCode(),
+				workflowDefinitionLink);
+	}
+
+	@Override
+	protected WorkflowDefinitionLink
+			testPutWorkflowDefinitionLinkByExternalReferenceCode_addWorkflowDefinitionLink()
+		throws Exception {
+
 		return workflowDefinitionLinkResource.
 			postWorkflowDefinitionWorkflowDefinitionLink(
 				_kaleoDefinition.getKaleoDefinitionId(),
-				workflowDefinitionLink);
+				randomWorkflowDefinitionLink());
+	}
+
+	@Override
+	protected WorkflowDefinitionLink
+			testPutWorkflowDefinitionLinkByExternalReferenceCode_createWorkflowDefinitionLink()
+		throws Exception {
+
+		return testPutWorkflowDefinitionLinkByExternalReferenceCode_addWorkflowDefinitionLink();
+	}
+
+	@Override
+	protected WorkflowDefinitionLink
+		testPutWorkflowDefinitionLinkByExternalReferenceCode_getWorkflowDefinitionLink(
+			String externalReferenceCode) {
+
+		try {
+			return _fetchWorkflowDefinitionLinkByExternalReferenceCode(
+				externalReferenceCode, _kaleoDefinition.getGroupId());
+		}
+		catch (PortalException portalException) {
+			return null;
+		}
 	}
 
+	private WorkflowDefinitionLink
+			_fetchWorkflowDefinitionLinkByExternalReferenceCode(
+				String externalReferenceCode, long groupId)
+		throws PortalException {
+
+		return _toWorkflowDefinitionLink(
+			_workflowDefinitionLinkLocalService.
+				fetchWorkflowDefinitionLinkByExternalReferenceCode(
+					externalReferenceCode, groupId));
+	}
+
+	private WorkflowDefinitionLink _toWorkflowDefinitionLink(
+			com.liferay.portal.kernel.model.WorkflowDefinitionLink
+				workflowDefinitionLink)
+		throws PortalException {
+
+		return new WorkflowDefinitionLink() {
+			{
+				setClassName(workflowDefinitionLink::getClassName);
+				setExternalReferenceCode(
+					workflowDefinitionLink::getExternalReferenceCode);
+				setGroupExternalReferenceCode(StringPool.BLANK);
```

Comment:
Why is this value hardcoded? Can we get the value from workflowDefinitionLink class like other fields do?